### DISCUSSION TITLE
How to handle CallbackQuery?

### DISCUSSION THREAD

**Discussion by Izntuser** [def-studio/telegraph#585] [Q&A]
Hi, there is smth strange and I cannot find out how to fix it. Will be happy for any reply.

I use Laravel 11 and Telegraph 1.44

Here is my inline keyboard:
`return Keyboard::make()->row([Button::make("Some Button")->action('dismiss')->param("value", 1)]);`
Keyboard successfully created, and I can see it on my Telegram bot chat.
<img width="472" alt="Знімок екрана 2024-05-07 о 23 25 10" src="https://github.com/defstudio/telegraph/assets/110115427/a2a9d079-47c4-4e88-ad03-290c5910d066">

But when I click it I see loading on that button and that's all...

My Handler:
```
class Handler extends WebhookHandler
{
    public function dismiss(): void
    {
        $this->reply("hello");
    }
    ...
}
```

My config:
`        'allow_callback_queries_from_unknown_chats' => true,`

**Comment by fabio-ivona** [comment]
Hi, is your webook correctly set up? You can use the debug config to see if any request reaches your server

**Reply by Izntuser** [reply]
Hi, thank you for the answer!

Yes, I can confirm that all my commands and other handlers working properly. 
But by clicking my "Some Button" I don't see any log, so callback query don't reach my server.

<img width="713" alt="Знімок екрана 2024-05-09 о 10 23 26" src="https://github.com/defstudio/telegraph/assets/110115427/505ef3f1-1c41-428c-af3f-042fe68f74e2">

**Reply by fabio-ivona** [reply]
could you put a log in eary part of Webhook handle method? I'm wondering in case an exception stops the execution too early. Telegram will keep trying to send the callback query endlessly in this case

**Reply by Izntuser** [reply]
You mean to add smth like that?
<img width="631" alt="Знімок екрана 2024-05-09 о 11 44 11" src="https://github.com/defstudio/telegraph/assets/110115427/6a26c6b7-85aa-4a72-976f-f5e532d4824d">


In that case, I also don't see any logs related to dismiss method, which should be triggered by clicking my "Some Button"

<img width="725" alt="Знімок екрана 2024-05-09 о 11 46 23" src="https://github.com/defstudio/telegraph/assets/110115427/d14e0973-f7b8-4539-a053-3d6ecb3201bc">

**Reply by fabio-ivona** [reply]
I mean even before, in the handle method of the parent webhook class

**Reply by Izntuser** [reply]
Aha, I see


I've added it here: 
`vendor/defstudio/telegraph/src/Controllers/WebhookController.php`
<img width="576" alt="Знімок екрана 2024-05-09 о 17 18 36" src="https://github.com/defstudio/telegraph/assets/110115427/a64f85d0-49cd-4c53-8a8c-2bb85da98233">


It works for text and commands, but it is not triggered by clicking "Some Button"

<img width="621" alt="Знімок екрана 2024-05-09 о 17 19 46" src="https://github.com/defstudio/telegraph/assets/110115427/93e6d966-6cc4-4111-9553-a4137fb60dde">

**Reply by Izntuser** [reply]
Okay, I'm sorry. I've created a new bot, and callback query works for it. 
But in any case, I'm wondering why it is not working for my first bot, do you have any idea?

**Reply by fabio-ivona** [reply]
hmm, nope, that's weird :(
